### ISSUE TITLE
Parentheses Missing

### ISSUE THREAD

**Issue by Somdy** [java-decompiler/jd-gui#400]
This may happen occasionally that decompiled codes are missing some parentheses. Not the parentheses that are parts of methods or constructors, but the ones in some calculation statements.  An example are as following.

This is decompiled by jd.
![图片](https://user-images.githubusercontent.com/45280869/186633696-01e4ecda-e2f3-451c-bfd9-0f9ea9043271.png)

This is decompiled by IDEA.
![图片](https://user-images.githubusercontent.com/45280869/186633873-5747b72e-2c6e-41cf-bb4f-fa43d6359477.png)

It is notable that a local var `handSizeAndDraw` is assigned by a statement `10 - this.amount + hand.size()` while the statement is `10 - (this.amount + hand.size())` in IDEA. The latter is correct after debugging it.

**Comment by Boat2017** [comment]
现在是假期。已经收到您的来信，我会尽快回复。

**Comment by StevenJack666** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by Somdy** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by rautamiekka** [comment]
`10 - this.amount + hand.size()`
`10 - (this.amount + hand.size())`

100% pointless in this case.

**Comment by Somdy** [comment]
> `10 - this.amount + hand.size()` `10 - (this.amount + hand.size())`
> 
> 100% pointless in this case.

@rautamiekka are u sure? i don't know what the case is you are saying and in my example 10-1+2=11 but 10-(1+2)=7

**Comment by L-ios** [comment]
can you show your class file

**Comment by Somdy** [comment]
> can you show your class file

@L-ios no. it's the code of a game on Steam called _Slay the Spire_ and i think it's copyrighted so i won't put the class file here. but i am sure the game would run in a completely wrong logic if the parentheses were missing.

**Comment by Somdy** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by rautamiekka** [comment]
> > `10 - this.amount + hand.size()` `10 - (this.amount + hand.size())`
> > 100% pointless in this case.
> 
> @rautamiekka are u sure? i don't know what the case is you are saying and in my example 10-1+2=11 but 10-(1+2)=7

_Sigh ... I hate math ..._
